Update StableMIR doc to reflect current status (#132085)
rust-lang/rust: compiler/stable_mir/README.md

This crate is currently developed in-tree together with the compiler.

Our goal is to start publishing `stable_mir` into crates.io.

> Until then, users will use this as any other rustc crate, via extern crate.

Reviewer: Perhaps mention that users need to install the `rustc-dev` component.

## Stable MIR Design

> The stable-mir will follow a similar approach to proc-macro2. It’s

Reviewer: ```suggestion
The stable-mir will follow a similar approach to proc-macro2. Its
```

> implementation is done using two main crates:

Reviewer: ```suggestion
implementation is split between two main crates:
```

> - `stable_mir`: Public crate, to be published on crates.io, which will contain

Reviewer: ```
- `stable_mir`: Public crate, to be published on crates.io, whics contains
the stable data structures and APIs to access them. It calls to `rustc_smir`.
- `rustc_smir`: This crate serves as `stable_mir`'s interface to the compiler.
It is responsible for gathering all the information requested, and converting
them to their stable data structures.
```

Reviewer: Github didn't allow me to rewrite this as an inline suggestion, but this section seems wrong. AFAICT, the `rustc_smir` crate is what does the conversion; hence it having all the `.stable` calls.

Author: Indeed. Not for long though, but I'll update this to match the current state. Thanks

Author: I documented that we are in a transition period. Let me know if that makes sense.
